robolectric/src/test/java/org/robolectric/shadows/ShadowKeyCharacterMapTest.java

```diff
@@ -0,0 +1,48 @@
+package org.robolectric.shadows;
+
+import android.view.KeyCharacterMap;
+import android.view.KeyEvent;
+import android.widget.EditText;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RuntimeEnvironment;
+import org.robolectric.TestRunners;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(TestRunners.MultiApiWithDefaults.class)
+public class ShadowKeyCharacterMapTest {
+
+  // This is an end-to-end test of key event dispatch which relies on
+  // the ShadowKeyCharacterMap at several points in the process.
+  @Test
+
+  public void test_dispatchKeyEvents() throws Exception {
+    test_dispatchKeyEvents( "string" );
```

Comment:
It's a little hard to tell what the test is testing, since it's all contained in a helper method. I would suggest removing the TODOs at the end of the method and in-lining `test_dispatchKeyEvents`.